pyiron_base/jobs/job/runfunction.py

```diff
@@ -431,6 +446,91 @@ def run_job_with_runmode_srun(job):
+        executor (concurrent.futures.Executor): executor class which implements the executor interface defined in the
+                                                python concurrent.futures.Executor class.
+        gpus_per_slot (int): number of GPUs per MPI rank, typically 1
+
+     Returns:
+         concurrent.futures.Future: future object to develop asynchronous simulation protocols
+    """
+
+    if static_isinstance(
+        obj=job, obj_type="pyiron_base.jobs.master.generic.GenericMaster"
+    ):
+        raise NotImplementedError(
+            "Currently job.server.run_mode.executor does not support GenericMaster jobs."
+        )
+    if isinstance(executor, ProcessPoolExecutor):
+        return run_job_with_runmode_executor_futures(job=job, executor=executor)
+    else:
+        raise NotImplementedError(
+            "Currently only flux.job.FluxExecutor and concurrent.futures.ProcessPoolExecutor are supported."
+        )
+
+
+def run_job_with_runmode_executor_futures(job, executor):
+    """
+    Interface for the ProcessPoolExecutor implemented in the python standard library as part of the concurrent.futures
+    module. The ProcessPoolExecutor does not provide any resource management, so the user is responsible to keep track of
+    the number of compute cores in use, as over-subscription can lead to low performance.
+
+    >>> from concurrent.futures import ProcessPoolExecutor
+    >>> job.server.executor = ProcessPoolExecutor()
+    >>> fs = job.run()
+
+    Args:
+        job (GenericJob): pyiron job object
+        executor (concurrent.futures.Executor): executor class which implements the executor interface defined in the
+                                                python concurrent.futures.Executor class.
+
+     Returns:
+         concurrent.futures.Future: future object to develop asynchronous simulation protocols
+    """
+    if not state.database.database_is_disabled:
+        if not state.database.using_local_database:
+            return executor.submit(
+                multiprocess_wrapper,
+                working_directory=job.project_hdf5.working_directory,
+                job_id=job.job_id,
+                file_path=None,
+                debug=False,
+                connection_string=None,
+            )
+        else:
+            return executor.submit(
+                multiprocess_wrapper,
+                working_directory=job.project_hdf5.working_directory,
+                job_id=job.job_id,
+                file_path=None,
+                debug=False,
+                connection_string=str(job.project.db.conn.engine.url),
+            )
+    else:
+        return executor.submit(
+            multiprocess_wrapper,
+            working_directory=job.project_hdf5.working_directory,
+            job_id=None,
+            file_path=job.project_hdf5.file_name + job.project_hdf5.h5_path,
+            debug=False,
+            connection_string=None,
+        )
```

Comment:
```suggestion
    if state.database.database_is_disabled:
        file_path = job.project_hdf5.file_name + job.project_hdf5.h5_path
        connection_string = None
    else:
        file_path = None
        if state.database.using_local_database:
            connection_string = str(job.project.db.conn.engine.url)
        else:
            connection_string = None
    
    return executor.submit(
                multiprocess_wrapper,
                working_directory=job.project_hdf5.working_directory,
                job_id=job.job_id,
                file_path=file_path,
                debug=False,
                connection_string=connection_string,
            )
```

Improve readability by factoring out the common code and avoiding logical negations

Reply:
I find the above to be the most readable, but here is a line-minimized version that I don't find too bad:

```python
    connection_string = None
    file_path = None

    if not state.database.database_is_disabled and state.database.using_local_database: 
        connection_string=str(job.project.db.conn.engine.url)
    elif state.database.database_is_disabled:
        file_path=job.project_hdf5.file_name + job.project_hdf5.h5_path
    return executor.submit(
                multiprocess_wrapper,
                working_directory=job.project_hdf5.working_directory,
                job_id=job.job_id,
                file_path=file_path,
                debug=False,
                connection_string=connection_string,
            )
```